CompilerSource/settings-parse/parse_ide_settings.cpp

```diff
@@ -141,7 +142,16 @@ void parse_ide_settings(const char* eyaml)
   setting::keyword_blacklist = settree.get("keyword-blacklist").toString();
 
   // The number of compile jobs
-  num_make_jobs = settree.get("jobs").toString();
+  if (settree.exists("jobs")) {
+    const std::string &jobs = settree.get("jobs");
+    if (std::all_of(jobs.begin(), jobs.end(), [](char c) { return std::isdigit(c); })) {
+      num_make_jobs = settree.get("jobs").toString();
+    } else {
+      num_make_jobs = "1";
+    }
+  } else {
+    num_make_jobs = "1";
```

Comment:
Might I suggest just setting `num_make_jobs = "1";` up front and allowing the main branch to override?